src/js/wrap.cpp

```diff
@@ -410,6 +411,131 @@ namespace js
     return JS_UNDEFINED;
   }
 
+  JSValue js_gov_set_jwt_public_signing_keys(
+    JSContext* ctx,
+    [[maybe_unused]] JSValueConst this_val,
+    int argc,
+    JSValueConst* argv)
+  {
+    if (argc != 3)
+    {
+      return JS_ThrowTypeError(ctx, "Passed %d arguments but expected 3", argc);
+    }
+
+    // yikes
+    auto global_obj = JS_GetGlobalObject(ctx);
+    auto ccf = JS_GetPropertyStr(ctx, global_obj, "ccf");
+    auto kv = JS_GetPropertyStr(ctx, ccf, "kv");
+
+    auto tx_ctx_ptr = static_cast<TxContext*>(JS_GetOpaque(kv, kv_class_id));
+
+    if (tx_ctx_ptr->tx == nullptr)
+    {
+      return JS_ThrowInternalError(ctx, "No transaction available");
+    }
+
+    JS_FreeValue(ctx, kv);
+    JS_FreeValue(ctx, ccf);
+    JS_FreeValue(ctx, global_obj);
+
+    auto& tx = *tx_ctx_ptr->tx;
+
+    auto issuer_cstr = JS_ToCString(ctx, argv[0]);
+    if (issuer_cstr == nullptr)
+    {
+      return JS_ThrowTypeError(ctx, "issuer argument is not a string");
+    }
+    std::string issuer(issuer_cstr);
+    JS_FreeCString(ctx, issuer_cstr);
+
+    JSValue metadata_val = JS_JSONStringify(ctx, argv[1], JS_NULL, JS_NULL);
+    if (JS_IsException(metadata_val))
+    {
+      return JS_ThrowTypeError(ctx, "metadata argument is not a JSON object");
+    }
+    auto metadata_cstr = JS_ToCString(ctx, metadata_val);
+    std::string metadata_json(metadata_cstr);
+    JS_FreeCString(ctx, metadata_cstr);
+    JS_FreeValue(ctx, metadata_val);
+
+    JSValue jwks_val = JS_JSONStringify(ctx, argv[2], JS_NULL, JS_NULL);
+    if (JS_IsException(jwks_val))
+    {
+      return JS_ThrowTypeError(ctx, "jwks argument is not a JSON object");
+    }
+    auto jwks_cstr = JS_ToCString(ctx, jwks_val);
+    std::string jwks_json(jwks_cstr);
+    JS_FreeCString(ctx, jwks_cstr);
+    JS_FreeValue(ctx, jwks_val);
+
+    try
+    {
+      auto metadata =
+        nlohmann::json::parse(metadata_json).get<ccf::JwtIssuerMetadata>();
+      auto jwks = nlohmann::json::parse(jwks_json).get<ccf::JsonWebKeySet>();
+      auto success =
+        ccf::set_jwt_public_signing_keys(tx, "<js>", issuer, metadata, jwks);
+      if (!success)
+      {
+        return JS_ThrowInternalError(
+          ctx, "set_jwt_public_signing_keys() failed");
+      }
+    }
+    catch (std::exception& exc)
+    {
+      return JS_ThrowInternalError(ctx, "Unexpected error: %s", exc.what());
```

Comment:
I think "Error" is fine, unexpected seems like a reasonable default assumption in an exception context.